examples/pipeline/wav2letter.py

```diff
@@ -0,0 +1,1179 @@
+    # input_lengths = tensors_lengths
+
+    # CTC
+    # outputs: input length, batch size, number of classes (including blank)
+    # targets: batch size, max target length
+    # input_lengths: batch size
+    # target_lengths: batch size
+
+    return criterion(outputs, targets, tensors_lengths, target_lengths)
+
+
+inds = random.sample(range(args.batch_size), k=2)
+
+
+def forward_decode(inputs, targets, decoder):
+
+    inputs = inputs.to(device, non_blocking=True)
+    output = model(inputs).to("cpu")
+    output = decoder(output)
+
+    output = decode(output.tolist())
+    target = decode(targets.tolist())
+
+    print_length = 20
+    for i in inds:
+        output_print = output[i].ljust(print_length)[:print_length]
+        target_print = target[i].ljust(print_length)[:print_length]
+        print(
+            f"Epoch: {epoch:4}   Target: {target_print}   Output: {output_print}", flush=True)
+
+    cers = [levenshtein_distance(a, b) for a, b in zip(target, output)]
+    cers_normalized = [d/len(a) for a, d in zip(target, cers)]
+    cers = statistics.mean(cers)
+    cers_normalized = statistics.mean(cers_normalized)
+
+    output = [o.split(char_space) for o in output]
+    target = [o.split(char_space) for o in target]
+
+    wers = [levenshtein_distance(a, b) for a, b in zip(target, output)]
+    wers_normalized = [d/len(a) for a, d in zip(target, wers)]
+    wers = statistics.mean(wers)
+    wers_normalized = statistics.mean(wers_normalized)
+
+    print(f"Epoch: {epoch:4}   CER: {cers:1.5f}   WER: {wers:1.5f}", flush=True)
+
+    return cers, wers, cers_normalized, wers_normalized
+
+history_loader = defaultdict(list)
+history_training = defaultdict(list)
+history_validation = defaultdict(list)
+
+if args.resume and os.path.isfile(CHECKPOINT_filename):
+    print("Checkpoint: loading '{}'".format(CHECKPOINT_filename))
+    checkpoint = torch.load(CHECKPOINT_filename)
+
+    args.start_epoch = checkpoint['epoch']
+    best_loss = checkpoint['best_loss']
+    history_training = checkpoint['history_training']
+    history_validation = checkpoint['history_validation']
+
+    model.load_state_dict(checkpoint['state_dict'])
+    optimizer.load_state_dict(checkpoint['optimizer'])
+    scheduler.load_state_dict(checkpoint['scheduler'])
+
+    print("Checkpoint: loaded '{}' at epoch {}".format(
+        CHECKPOINT_filename, checkpoint['epoch']))
+    print(tabulate(history_training, headers="keys"), flush=True)
+    print(tabulate(history_validation, headers="keys"), flush=True)
+else:
+    print("Checkpoint: not found")
+
+    save_checkpoint({
+        'epoch': args.start_epoch,
+        'state_dict': model.state_dict(),
+        'best_loss': best_loss,
+        'optimizer': optimizer.state_dict(),
+        'scheduler': scheduler.state_dict(),
+        'history_training': history_training,
+        'history_validation': history_validation,
+    }, False)
+
+
+with tqdm(total=args.epochs, unit_scale=1, disable=args.distributed) as pbar:
```

Comment:
Use of `tqdm` prevents to run the script in background while keeping the log in readable manner. I'd not use it.

Reply:
This could be made a flag for args (e.g. --verbose)